File: app/src/main/java/ch/admin/bag/dp3t/onboarding/OnboardingFinishedFragment.java
/*
 * Created by Ubique Innovation AG
 * https://www.ubique.ch
 * Copyright (c) 2020. All rights reserved.
 */
package ch.admin.bag.dp3t.onboarding;

import android.os.Bundle;
import android.view.View;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import ch.admin.bag.dp3t.R;

public class OnboardingFinishedFragment extends Fragment {

	public static OnboardingFinishedFragment newInstance() {
		return new OnboardingFinishedFragment();
	}

	public OnboardingFinishedFragment() {
		super(R.layout.fragment_onboarding_finished);
	}

	@Override
	public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
		view.findViewById(R.id.onboarding_continue_button)
				.setOnClickListener(v -> {
					((OnboardingActivity) getActivity()).continueToNextPage();
				});
	}

}

File: app/src/main/java/ch/admin/bag/dp3t/onboarding/OnboardingFinishedFragment.kt
/*
 * Copyright (c) 2020 Ubique Innovation AG <https://www.ubique.ch>
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * SPDX-License-Identifier: MPL-2.0
 */
package ch.admin.bag.dp3t.onboarding

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.os.bundleOf
import androidx.fragment.app.Fragment
import ch.admin.bag.dp3t.R
import ch.admin.bag.dp3t.databinding.FragmentOnboardingFinishedBinding

private const val ARG_ONBOARDING_TYPE = "ARG_ONBOARDING_TYPE"

class OnboardingFinishedFragment : Fragment() {

	companion object {
		fun newInstance(onboardingType: OnboardingType) = OnboardingFinishedFragment().apply {
			arguments = bundleOf(ARG_ONBOARDING_TYPE to onboardingType)
		}
	}

	override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
		return FragmentOnboardingFinishedBinding.inflate(inflater).apply {
			val onboardingType = requireArguments().getSerializable(ARG_ONBOARDING_TYPE) as OnboardingType
			if (onboardingType == OnboardingType.NON_INSTANT_PART) {
				onboardingTitle.setText(R.string.partial_onboarding_done_title)
				onboardingText.setText(R.string.partial_onboarding_done_text)
				onboardingContinueButton.setText(R.string.partial_onboarding_box_action)
			}
			onboardingContinueButton.setOnClickListener { (activity as OnboardingActivity).continueToNextPage() }
		}.root
	}

}